olive/model/handler/mixin/hf_config.py

```diff
@@ -44,6 +50,63 @@ def get_hf_model_config(self):
 
         return get_hf_model_config(self._get_model_path_or_name(), **self.hf_config.get_loading_args_from_pretrained())
 
+    def save_hf_model_config(self, output_dir: str, **kwargs):
+        if self.hf_config is None:
+            raise ValueError("HF model_config is not available.")
+        if not Path(output_dir).is_dir():
+            raise ValueError("Expecting a directory as input.")
+
+        save_hf_model_config(self.get_hf_model_config(), output_dir, **kwargs)
+
+    def get_hf_model_generation_config(self):
+        if self.hf_config is None:
+            raise ValueError("HF model_config is not available")
+
+        return get_hf_model_generation_config(
+            self._get_model_path_or_name(), **self.hf_config.get_loading_args_from_pretrained()
+        )
+
+    def save_hf_model_generation_config(self, output_dir: str, **kwargs):
+        if self.hf_config is None:
+            raise ValueError("HF model_config is not available.")
+        if not Path(output_dir).is_dir():
+            raise ValueError("Expecting a directory as input.")
+
+        save_hf_model_generation_config(self.get_hf_model_generation_config(), output_dir, **kwargs)
+
+    def get_hf_model_tokenizer(self):
+        if self.hf_config is None:
+            raise ValueError("HF model_config is not available")
+
+        return get_hf_model_tokenizer(
+            self._get_model_path_or_name(), **self.hf_config.get_loading_args_from_pretrained()
+        )
+
+    def save_hf_model_tokenizer(self, output_dir: str, **kwargs):
+        if self.hf_config is None:
+            raise ValueError("HF model_config is not available.")
+        if not Path(output_dir).is_dir():
+            raise ValueError("Expecting a directory as input.")
+
+        save_hf_model_tokenizer(self.get_hf_model_tokenizer(), output_dir, **kwargs)
+
+    def save_metadata_for_token_generation(self, output_dir: str, **kwargs):
```

Comment:
it seems only this method need to be public?

Reply:
All save_XXX functions removed in follow up PR.